### PR TITLE
NN-3052 Add moved in by to current location on cell history

### PR DESCRIPTION
**Pull request by jsrobertson**
- Add Moved in by to current location
- Re ordered and re-labeled headings in the cell history tables
- Refactored a bit of the code (see comments)

image

### CODE REVIEW

File: backend/controllers/prisonerProfile/prisonerCellHistory.js [hunk 1]
```diff
@@ -32,46 +32,39 @@ module.exports = ({ oauthApi, prisonApi, logError, page = 0 }) => async (req, re
...
-          const staffDetails = await prisonApi.getStaffDetails(res.locals, cell.movementMadeBy)
-          const agency = agencyData.find(agencyDetails => cell.agencyId === agencyDetails.agencyId)
-
-          return {
-            establishment: agency.description,
-            location: extractLocation(cell.description, cell.agencyId),
-            movedIn: cell.assignmentDateTime && formatTimestampToDateTime(cell.assignmentDateTime),
-            movedOut: cell.assignmentEndDateTime && formatTimestampToDateTime(cell.assignmentEndDateTime),
-            assignmentDateTime: moment(cell.assignmentDateTime).format('YYYY-MM-DDTHH:mm:ss'),
-            assignmentEndDateTime: cell.assignmentEndDateTime
```
Comment (jsrobertson) [inline backend/controllers/prisonerProfile/prisonerCellHistory.js:62]: Checking for `cell.assignmentEndDateTime` but they've already been filtered out above.

File: backend/controllers/prisonerProfile/prisonerCellHistory.js [hunk 2]
```diff
@@ -82,18 +75,7 @@ module.exports = ({ oauthApi, prisonApi, logError, page = 0 }) => async (req, re
             }
           })
         : [],
-      currentLocation: {
-        establishment: agencyData.find(agencyDetails => currentLocation.agencyId === agencyDetails.agencyId)
-          .description,
-        location: extractLocation(currentLocation.description, currentLocation.agencyId),
-        movedIn: currentLocation.assignmentDateTime && formatTimestampToDateTime(currentLocation.assignmentDateTime),
-        assignmentDateTime: moment(currentLocation.assignmentDateTime).format('YYYY-MM-DDTHH:mm:ss'),
-        assignmentEndDateTime: currentLocation.assignmentEndDateTime
```
Comment (jsrobertson) [inline backend/controllers/prisonerProfile/prisonerCellHistory.js:91]: Again, checking for `cell.assignmentEndDateTime` but `currentLocation` is the cell which doesn't have an `assignmentEndDateTime`, so not needed.

File: backend/controllers/prisonerProfile/prisonerCellHistory.js [hunk 3]
```diff
@@ -82,18 +75,7 @@ module.exports = ({ oauthApi, prisonApi, logError, page = 0 }) => async (req, re
...
-          ? moment(currentLocation.assignmentEndDate).format('YYYY-MM-DDTHH:mm:ss')
-          : today,
-        livingUnitId: currentLocation.livingUnitId,
-        agencyId: currentLocation.agencyId,
-      },
+      currentLocation,
```
Comment (jsrobertson) [inline backend/controllers/prisonerProfile/prisonerCellHistory.js:78]: `currentLocation` now gets it's structure from `cellData` which removed a bit of duplication.

**Review by JamieDraperUK — APPROVED** [review]
:boom: LGTM :boom:

**Pull request merged**